include/modem/nrf_modem_lib.h

```diff
@@ -38,27 +38,37 @@ enum nrf_modem_mode {
 };
 
 /**
- * @brief Initialize the Modem library and turn on the modem.
+ * @brief Initialize the Modem library in normal operating mode and turn on the modem.
  *
- * This function initializes all integration components of the nrf_modem library
- * and turns on the modem. The operation can take a few minutes when a firmware update
- * is scheduled. If your application supports modem firmware updates, consider initializing
- * the library manually to have control of what the application should do in the meantime.
+ * This function initializes the modem in normal operating mode and initializes all integration
+ * components of the Modem library and turns on the modem. The operation can take a few minutes
+ * when a firmware update is scheduled.
  *
- * The library can initialize the modem in normal mode or bootloader mode.
+ * To switch between the bootloader mode and normal operating mode, shutdown the modem
+ * with @ref nrf_modem_lib_shutdown() and re-initialize it in the desired mode.
+ * Use @ref nrf_modem_lib_init() to initialize in normal mode and
+ * @ref nrf_modem_lib_bootloader_init() to initialize the Modem library in bootloader mode.
  *
- * The bootloader mode is used to update the whole modem firmware.
- * When modem is initialized in bootloader mode, no other functionality is available.
- * In particular, networking sockets and AT commands won't be available.
+ * @return int Zero on success, non-zero otherwise.
+ */
+int nrf_modem_lib_init(void);
+
+/**
+ * @brief Initialize the Modem library in bootloader mode and turn on the modem.
  *
- * To switch between the bootloader mode and normal mode, shutdown the modem
- * with @ref nrf_modem_lib_shutdown() and re-initialize it in the desired mode.
+ * This function initializes the components of the Modem library necessary for performing
+ * full modem firmware updates and turns on the modem. When the modem is initialized in bootloader
+ * mode, no other functionality is available. In particular, networking sockets and AT commands
+ * won't be available.
```

Comment:
First part is a bit redundant, I would start directly with `When the modem is initialized in bootloader ..` and omit the first part.